mne/decoding/time_gen.py

```diff
@@ -1,14 +1,984 @@
+        if "length" in self:
+            s += ", length: %s s" % (self["length"])
+        if "slices" in self:
+            # identify depth: training times only contains n_time but
+            # testing_times can contain n_times or n_times * m_times
+            depth = [len(ii) for ii in self["slices"]]
+            if len(np.unique(depth)) == 1:  # if all slices have same depth
+                if depth[0] == 1:  # if depth is one
+                    s += ", n_time_windows: %s" % (len(depth))
+                else:
+                    s += ", n_time_windows: %s * %s" % (len(depth), depth[0])
+            else:
+                s += (", n_time_windows: %s * [%s, %s]" %
+                      (len(depth),
+                       min([len(ii) for ii in depth]),
+                       max(([len(ii) for ii in depth]))))
+        return "<DecodingTime | %s>" % s
+
+
+class GeneralizationAcrossTime(object):
+    """Generalize across time and conditions
+
+    Creates and estimator object used to 1) fit a series of classifiers on
+    multidimensional time-resolved data, and 2) test the ability of each
+    classifier to generalize across other time samples.
+
+    Parameters
+    ----------
+    clf : object | None
+        An estimator compliant with the scikit-learn API (fit & predict).
+        If None the classifier will be a standard pipeline including
+        StandardScaler and a linear SVM with default parameters.
+    cv : int | object
+        If an integer is passed, it is the number of folds.
+        Specific cross-validation objects can be passed, see
+        sklearn.cross_validation module for the list of possible objects.
+        Defaults to 5.
+    train_times : dict | None
+        A dictionary to configure the training times.
+        'slices' : np.ndarray, shape (n_clfs,)
+            Array of time slices (in indices) used for each classifier.
+            If not given, computed from 'start', 'stop', 'length', 'step'.
+        'start' : float
+            Time at which to start decoding (in seconds). By default,
+            min(epochs.times).
+        'stop' : float
+            Maximal time at which to stop decoding (in seconds). By default,
+            max(times).
+        'step' : float
+            Duration separating the start of subsequent classifiers (in
+            seconds). By default, equals one time sample.
+        'length' : float
+            Duration of each classifier (in seconds). By default, equals one
+            time sample.
+        If None, empty dict. Defaults to None.
+    predict_type : str, optional, {'predict', 'predict_proba',
+        'decision_function'}
+        Indicates the type of prediction:
+            'predict' : generates a categorical estimate of each trial.
+            'predict_proba' : generates a probabilistic estimate of each trial.
+            'decision_function' : generates a continuous non-probabilistic
+                estimate of each trial.
```

Comment:
Do these look okay in the doc builds? They might need an extra newline before the new section and/or bullets